server/src/error.rs

```diff
@@ -1,17 +1,49 @@
 #![allow(clippy::module_name_repetitions)]
+
 use axum::{
     http::{Response, StatusCode},
     response::IntoResponse,
 };
 use thiserror::Error;
 use tracing::{event, Level};
 
+use crate::auth::error::{LoginError, RegisterError, VerifyError};
+use crate::auth::middleware::AuthorizationError;
+
 #[derive(Error, Debug)]
 pub enum CheeseError {
     #[error("MongoDB encountered an error")]
     Mongo(#[from] mongodb::error::Error),
+
     #[error("Encountered an ObjectID error")]
     Oid(#[from] mongodb::bson::oid::Error),
+
+    #[error("login")]
+    Login(#[from] LoginError),
+    #[error("register")]
+    Register(#[from] RegisterError),
+
+    #[error("authorization")]
+    Authorization(#[from] AuthorizationError),
+}
+
+macro_rules! response_error {
```

Comment:
A comment on how to use the macro might be nice

Reply:
fixed in latest commit